middleware.ts

```diff
@@ -0,0 +1,17 @@
+import { NextRequest } from 'next/server'
+import { NextResponse } from 'next/server'
+
+// This function can be marked `async` if using `await` inside
+export const middleware = (request: NextRequest) => {
+  if(request.nextUrl.pathname === '/articles'){
+    return NextResponse.rewrite(new URL('/articles/category/all', request.url));
+  }
+  if(request.nextUrl.pathname.endsWith('category/all')){
+    return NextResponse.redirect(new URL('/articles', request.url));
```

Comment:
So for example `/products/category/all` would be redirected to `/articles`?

Reply:
Why is this redirect (I mean `/articles/category/all` -> `/articles`) even necessary? I thought we want to redirect the other way around.